tracer/src/Datadog.Trace/ISpanContextExtractor.cs

```diff
@@ -26,8 +26,10 @@ public interface ISpanContextExtractor
         /// </summary>
         /// <param name="carrier">The carrier of the SpanContext. Often a header (http, kafka message header...)</param>
         /// <param name="getter">Given a key name, returns values from the carrier</param>
+        /// <param name="messageType">Use for Data Streams Monitoring: The type of messaging system where the message is coming from.</param>
+        /// <param name="source">Use for Data Streams Monitoring: The queue or topic where the message is coming from.</param>
         /// <typeparam name="TCarrier">Type of the carrier</typeparam>
         /// <returns>A potentially null Datadog SpanContext</returns>
-        public ISpanContext? Extract<TCarrier>(TCarrier carrier, Func<TCarrier, string, IEnumerable<string?>> getter);
+        public ISpanContext? Extract<TCarrier>(TCarrier carrier, Func<TCarrier, string, IEnumerable<string?>> getter, string? messageType = null, string? source = null);
```

Comment:
Is this a breaking change to a public API?

Reply:
Couldn't `messageType` and `source` be exposed as key/value pairs like all other values in `getter` instead of adding them as explicit parameters?

Reply:
Thanks for clarifying.